great_expectations/checkpoint/checkpoint.py

```diff
@@ -365,6 +359,20 @@ def run(  # noqa: C901, PLR0913, PLR0915
             checkpoint_config=self.config,
         )
 
+    @staticmethod
+    def _convert_validations_list_to_checkpoint_validation_configs(
+        validations: list[dict] | list[CheckpointValidationConfig] | None,
+    ) -> list[CheckpointValidationConfig]:
+        # We accept both dicts and rich config types but all internal usage should use the latter
+        if not validations:
+            return []
+        return [
+            CheckpointValidationConfig(**validation)
+            if isinstance(validation, dict)
+            else validation
+            for validation in validations
+        ]
```

Comment:
(non-blocking): There are 3 instances of this block of code in this PR. Is there a way to make this more DRY?

Reply:
Yeah trying to figure out a better way but as long as this is incrementally better, I'd push to make these changes and then continue iterating with subsequent PRs - finding it quite difficult to refactor given how deeply entrenched some of these typing issues are